tests/test-sessions/test-sessions-common/src/test/java/org/eclipse/jetty/server/session/AbstractSessionCacheTest.java

```diff
@@ -43,6 +48,62 @@
  */
 public abstract class AbstractSessionCacheTest
 {
+    public static class UnreadableSessionDataStore extends AbstractSessionDataStore
+    {
+        public Set<String> _deletedIds = ConcurrentHashMap.newKeySet();
```

Comment:
Perhaps you can just use `_data` in the implementation, so you won't need to remember deleted ids.